newton/_src/sim/builder.py

```diff
@@ -4438,20 +4437,32 @@ def find_shape_contact_pairs(self, model: Model):
         filters = copy.copy(self.shape_collision_filter_pairs)
         contact_pairs = []
 
-        # iterate over collision groups (islands)
-        for group, shapes in self.shape_collision_group_map.items():
```

Comment:
We should be able to remove `shape_collision_group_map` from the builder and model now.